### PR TITLE
Calculator: Milestone 5

### PR DESCRIPTION
**Pull request by gurbuzertunga**
In this branch:

- I updated App to Class component,
- Implemented calculation to the app

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello @gurbuzertunga  👋

**You've done pretty well here. 👌**

However, let's make a couple of changes.

- [x] After clicking an operator and then another number the display doesn't reflect the number clicked. Kindly ensure the number clicked after an operator is displayed.

- [x] In addition, kindly handle the error that results from a division by zero.

- [x] Optional - Consider reducing the font-size of the button text. It's currently pretty large.

- Please refer to other comments on the code.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

File: src/components/button.js [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
+const Button = ({ name, property, clickHandler }) => (
+  <button type="button" className={property} onClick={() => clickHandler(name)}>
     {name}
   </button>
 );
 
 Button.propTypes = {
   name: PropTypes.string,
   property: PropTypes.string,
+  clickHandler: PropTypes.func.isRequired,
```
Comment (mikenath223) [inline src/components/button.js:12]: - [ ] Optional - Consider changing the component filename from `button.js` to `Button.js`. It's a convention to capitalize components filenames.

File: src/components/buttonPanel.js [hunk 2]
```diff
@@ -1,37 +1,42 @@
...
 
-const ButtonPanel = () => (
+const ButtonPanel = ({ clickHandler }) => (
   <div className="button-container">
     <div className="top-buttons">
-      <Button name="AC" property="basic-op" />
-      <Button name="+/-" property="basic-op" />
-      <Button name="%" property="basic-op" />
-      <Button name="/" property="basic-op" />
+      <Button name="AC" property="basic-op" clickHandler={clickHandler} />
```
Comment (mikenath223) [inline src/components/buttonPanel.js:7]: - [ ] Optional - Consider changing the component filename from `buttonPanel.js` to `ButtonPanel.js`

**Comment by mikenath223** [comment]
- [x] Optional - Also consider changing the component filename of `display` from `display.js` to `Display.js`

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello @gurbuzertunga  👋

**You've done pretty well here fixing issues raised in the previous review. 👌**

However, let's make these changes.

- [x] There are two linting issues in `src/commponents/App.js`. Kindly fix them.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

**Review by mosaaleb — APPROVED** [review]
#### Milestone Approved :heavy_check_mark: 

Well done @gurbuzertunga :clap:. You have done all the requirements from the previous milestone. All linter issues are now fixed.

You can merge :+1: 

##### Muhammad Ebeid

**Pull request merged**